scipy/stats/tests/test_qmc.py

```diff
@@ -374,31 +374,76 @@ def random(self, n=1):
         return sample
 
 
-def test_subclassing_QMCEngine():
-    engine = RandomEngine(2, seed=175180605424926556207367152557812293274)
+class TestQmcEngine:
```

Comment:
This made the diff messy. Please put the non-essential changes (like putting this all in a class, if that's what you're after) in a separate PR to speed up review overall. I'll review the tests when that's done.